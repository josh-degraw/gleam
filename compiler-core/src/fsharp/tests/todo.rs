use crate::assert_fsharp;

#[test]
fn plain() {
    assert_fsharp!(
        r#"
pub fn main() {
  todo
}
"#
    );
}

#[test]
fn todo_as() {
    assert_fsharp!(
        r#"
pub fn main() {
  todo as "wibble"
}
"#
    );
}

#[test]
fn todo_as_function() {
    assert_fsharp!(
        r#"
pub fn retstring() {
  "wibble"
}
pub fn main() {
  todo as { retstring() <> "wobble" }
}
"#
    );
}

#[test]
fn piped() {
    assert_fsharp!(
        r#"
     pub fn main() {
      "lets"
      |> todo as "pipe"
      |> todo as "other todo"
    }
    "#
    );
}
