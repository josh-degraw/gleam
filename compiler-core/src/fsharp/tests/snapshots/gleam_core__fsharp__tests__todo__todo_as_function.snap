---
source: compiler-core/src/fsharp/tests/todo.rs
assertion_line: 27
expression: "\npub fn retstring() {\n  \"wibble\"\n}\npub fn main() {\n  todo as { retstring() <> \"wobble\" }\n}\n"
---
module rec my.``mod``
let retstring(): string = begin
    "wibble"
end
let main(): 't10 = begin
    failwith (retstring() + "wobble")
end
